sklearn/tests/test_pipeline.py

```diff
@@ -603,6 +603,67 @@ def test_make_union_kwargs():
         make_union(pca, mock, transformer_weights={"pca": 10, "Transf": 1})
 
 
+def create_mock_transformer(base_name, n_features=3):
+    """Helper to create a mock transformer with custom feature names."""
+    mock = Transf()
+    mock.get_feature_names_out = lambda input_features: [
+        f"{base_name}{i}" for i in range(n_features)
+    ]
+    return mock
+
+
+def test_make_union_verbose_feature_names_out_enabled():
```

Comment:
Since, we already have similar test for `FeatureUnion`. I think it's enough to test that `make_union` sets `verbose_feature_names_out` correctly. Something like this:

```python
def test_make_union_passes_verbose_feature_names_out():
    union = make_union(..., verbose_feature_names_out=False)
    assert not union.verbose_feature_names_out
```

Reply:
Hi @thomasjpfan,

I've simplified the test case in my recent commit. Could you please review it when you get a chance?

Thanks!